pilot/pkg/model/push_context.go

```diff
@@ -128,28 +128,28 @@ func newDestinationRuleIndex() destinationRuleIndex {
 
 // sidecarIndex is the index of sidecar rules
 type sidecarIndex struct {
-	// sidecars for each namespace
+	// user configured sidecars for each namespace if available.
 	sidecarsByNamespace map[string][]*SidecarScope
 	// the Sidecar for the root namespace (if present). This applies to any namespace without its own Sidecar.
 	rootConfig *config.Config
-	// computedSidecarsByNamespace contains the default sidecar for namespaces that do not have a sidecar.
-	// These may be DefaultSidecarScopeForNamespace if rootConfig is empty or ConvertToSidecarScope if not.
+	// rootConfigSidecarsByNamespace contains the default sidecar for namespaces that do not have a sidecar.
+	// These are converted from rootConfig if it exists.
 	// These are lazy-loaded. Access protected by defaultSidecarMu
-	computedSidecarsByNamespace map[string]*SidecarScope
-	// gatewayDefaultSidecarsByNamespace contains the default sidecar for namespaces that do not have a sidecar,
-	// for gateways.
-	// Unlike computedSidecarsByNamespace, this is *always* the output of DefaultSidecarScopeForNamespace.
+	rootConfigSidecarsByNamespace map[string]*SidecarScope
```

Comment:
root is also confusing to make me think this is from `istio-system`

Reply:
it is not istio-system. It can be any thing that is configured as rootNamespace in mesh-config